docs/src/developer/e2e/runtimetest.md

```diff
@@ -0,0 +1,50 @@
+
+## Notes
+
+This binary must be compiled with the option of static linking to crt0 given to the rustc. If compiled without it, it will add a linking to /lib64/ld-linux-x86-64.so . The binary compiled this way cannot be run inside the container process, as they do not have access to /lib64/... Thus the runtime test must be statically linked to crt0.
+
+While developing, originally this was added to the common workspace of all crates in youki. But then it was realized that this was quite inefficient because :
+
+- All packages except runtimetest will be compiled with dynamic linking
+- Runtimetest will be compiled with static linking
+
+Now runtimetest needs at least `oci-spec` and `nix` package for its operations, which are also dependencies of other packages in the workspace. Thus both of these, and recursively their dependencies must be compiled twice, each time, once for dynamic linking and once for static. The took a long time in the compilation stage, especially when developing / adding new tests. Separating runtimetest from the workspace allows it to have a separate target/ directory, where it can store the statically compiled dependencies, and the workspace can have its target/ directory, where it can store its dynamically compiled dependencies. That way only the crates which have changes need to be compiled (runtimetest or integration test), and not their dependencies.
+
+In case in future this separation is not required, or some other configuration is chosen, make sure the multiple compilation issue does not arise, or the advantages of new method outweigh the time spent in double compilation.
+
+To see if a binary can be run inside the container process, run
+
+```console
+readelf -l path/to/binary |grep "program interpreter"
+```
+
+`[Requesting program interpreter: /lib64/ld-linux-x86-64.so.2]` means that the binary is not statically linked, and cannot be run inside the container process. If the above command gives no output, that means it does not require any program interpreter and can be run inside the container.
+
+Another way is to run
+
+```console
+file path/to/binary
+```
+
+```console
+./youki: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=...., for GNU/Linux 3.2.0, with debug_info, not stripped`
+```
+
+This output indicates that the binary is dynamically linked, thus cannot be run inside the container process
+
+```console
+./runtimetest: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), statically linked, BuildID[sha1]=...., for GNU/Linux 3.2.0, with debug_info, not stripped
+```
+
+This output indicates that the binary is statically linked, and can be run inside the container process
```

Comment:
Can we have the lines `This output indicates...` before their respective console examples? This way can be a bit confusing.

Reply:
Thanks for your comment, But it is only copied from the original. We have to take care of it in another PR.
https://containers.github.io/youki/developer/runtimetest.html

Reply:
ohk, we can do it separately then :+1: